### PR TITLE
update changelog for all 0.12.x releases

### PR DESCRIPTION
**Pull request by sritchie**
Plus some README updates.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,18 +64,7 @@ A list of contributors to the project can be found here: [Contributors](https://
...
-* `algebird-core_2.10`
-* `algebird-test_2.11`
-* `algebird-test_2.10`
-* `algebird-util_2.11`
-* `algebird-util_2.10`
-* `algebird-bijection_2.11`
-* `algebird-bijection_2.10`
-
-The suffix denotes the scala version.
+See [Algebird's page on the Scaladex](https://index.scala-lang.org/twitter/algebird) for information on all published artifacts and their associated Scala versions. Algebird currently supports Scala 2.10 and 2.11.
```
Comment (sritchie) [inline README.md]: they've got dropdowns for all combinations on the top right of the page... so good, and way easier to keep relevant.

File: README.md [hunk 2]
```diff
@@ -85,6 +75,8 @@ The suffix denotes the scala version.
 - [Packetloop](https://www.packetloop.com) (see [this tweet](https://twitter.com/cloudjunky/status/355073917720858626)
 - Ebay uses Algebird for machine learning: [ScalaDays talk](http://www.slideshare.net/VitalyGordon/scalable-and-flexible-machine-learning-with-scala-linkedin)
 
+Other projects built with Algebird, as compiled by the Scaladex: [![Scaladex Dependents](https://index.scala-lang.org/count.svg?q=dependencies:twitter/algebird*&subject=scaladex:&color=blue&style=flat-square)](https://index.scala-lang.org/count.svg?q=dependencies:twitter/algebird*)
```
Comment (sritchie) [inline README.md]: we've got a few!

**Comment by sritchie** [comment]
@johnynek, looks like you've been adding great docs to the Releases section! I'll look at that next time :)

File: README.md [hunk 3]
```diff
@@ -38,15 +38,16 @@ In the above, the class `Max[T]` signifies that the `+` operator should actually
...
+The latest API docs are hosted on Algebird's [ScalaDoc index](https://twitter.github.io/algebird/api/).
 
 ## Get Involved + Code of Conduct
 
-Pull requests and bug reports are always welcome!
+Pull requests and bug reports are always welcome! Check out our [Contributing guide](https://twitter.github.io/algebird/contributing.html) for information on what we most need help with and how you can get started contributing.
+
+Discussion occurs primarily on the Algebird Gitter channel: [![Chat](https://badges.gitter.im/twitter/algebird.svg)](https://gitter.im/twitter/algebird?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-Discussion occurs primarily on the [Algebird mailing list](https://groups.google.com/forum/#!forum/algebird).
```
Comment (johnynek) [inline README.md]: Should we remove the mailing list entirely? Seems like some might still prefer it.

Comment (sritchie) [inline README.md]: good call, added it back.

**Comment by sritchie** [comment]
@johnynek, added it back https://github.com/twitter/algebird/pull/590/commits/3b126efeb3fe34095d6832c735d58012123d6d86

**Pull request merged**